File: src/main/java/dev/alexnader/framed/block/frame/StairsFrame.java
package dev.alexnader.framed.block.frame;

import dev.alexnader.framed.block.entity.FrameBlockEntity;
import net.minecraft.block.Block;
import net.minecraft.block.BlockState;
import net.minecraft.block.Blocks;
import net.minecraft.block.StairsBlock;
import net.minecraft.block.entity.BlockEntity;
import net.minecraft.world.BlockView;

import javax.annotation.Nullable;

import static dev.alexnader.framed.Framed.BLOCK_ENTITY_TYPES;
import static dev.alexnader.framed.Framed.META;

public class StairsFrame extends StairsBlock implements net.minecraft.block.BlockEntityProvider, Frame {
    public StairsFrame(final BlockState baseBlockState, final Settings settings) {
        super(baseBlockState, settings);
    }

    @Override
    public @Nullable BlockEntity createBlockEntity(final BlockView world) {
        return BLOCK_ENTITY_TYPES.STAIRS_FRAME.instantiate();
    }
}

File: src/main/kotlin/dev/alexnader/framity/blocks/StairsFrame.kt
package dev.alexnader.framity.blocks

import dev.alexnader.framity.BLOCK_FRAME
import dev.alexnader.framity.STAIRS_FRAME
import dev.alexnader.framity.STAIRS_FRAME_ENTITY
import dev.alexnader.framity.block_entities.FrameEntity
import net.minecraft.block.Block
import net.minecraft.block.BlockEntityProvider
import net.minecraft.block.BlockState
import net.minecraft.block.StairsBlock
import net.minecraft.state.StateManager
import net.minecraft.world.BlockView

class StairsFrame : StairsBlock(BLOCK_FRAME.value.defaultState, FRAME_SETTINGS), BlockEntityProvider {
    override fun createBlockEntity(world: BlockView?) = FrameEntity(STAIRS_FRAME, STAIRS_FRAME_ENTITY)

    @Suppress("unused")
    override fun appendProperties(builder: StateManager.Builder<Block, BlockState>?) {
        super.appendProperties(builder)
    }
}